### ISSUE TITLE
Wished-characters not claiming.

### ISSUE THREAD

**Issue by Divontae0** [chiyadev/MudaeFarm#152]
Hi, im new to this but the characters written down inside the wished characters channel does not claim in the server i am in. My server has many bot channels for mudae. One of my character got rolled and it didn't auto claim.

Is this how its meant to be laid out by any chance? 

![Capture](https://user-images.githubusercontent.com/66707720/84220892-d82afb80-aacb-11ea-9485-f49ab364986a.PNG)

![Capture2](https://user-images.githubusercontent.com/66707720/84220895-d8c39200-aacb-11ea-8250-845ffe0f6150.PNG)

**Comment by oobohP** [comment]
@Divontae0 Did you copy and paste your server's **_Channel ID?_** in the _bot-channels_ channel?

If you don't know how to get your servers Channel ID, please refer to [Here](https://support.discord.com/hc/en-us/articles/206346498-Where-can-I-find-my-User-Server-Message-ID-)

**Comment by WaterBottled** [comment]
Im also having this problem

**Comment by Divontae0** [comment]
Yeah same everything else works except the claiming section idk why tbh unless im typing it wrong or something

**Comment by pinyo6** [comment]
enable custom emotes might do the trick, most servers use custom ones so

**Comment by WaterBottled** [comment]
i have custom emotes on true aswell
ill try to grab the logs next time someone i wish is rolled

**Comment by WaterBottled** [comment]
ok so my verbose logs are non-stop spamming this
![verbose logs](https://user-images.githubusercontent.com/66806027/84577880-18210580-ad8e-11ea-8710-aafb67700857.PNG)

**Comment by WaterBottled** [comment]
also it says this in the notepad logs
![verbosse log2](https://user-images.githubusercontent.com/66806027/84577948-a0070f80-ad8e-11ea-8f20-638e5b6cc4f4.png)

**Comment by WaterBottled** [comment]
the channels are linked in the bot channels and the bot picks up when someone does $ima and says it ignored the character but doesnt say anything when someones does a $w, $h or $m.
![normallogs](https://user-images.githubusercontent.com/66806027/84578056-9af69000-ad8f-11ea-8314-8a76396f3e3a.png)

**Comment by Tookie111** [comment]
same thing is happening to me, doesnt collect ones on my wishlist but does everything else perfectly

**Comment by WaterBottled** [comment]
ok so i realized that it has something to do with the server im in. as it picks up $w and the other rolls on other servers. could it be something to do with being in private channels that only certain people can see? or maybe some type of configuration of the channel? idk i dont know the slightest thing about coding really

**Comment by KingJackFrost** [comment]
I can attest that it is not a private channel issue. As long as you, the user, can do regular things in the private channel, it should not be an issue; it worked in my server's private channel.

**Comment by thebatsyjoke** [comment]
I found out what the issue was. In MudaeClaimer.cs, there is an exception to ignore all $im messages by ignoring messages which contain "claims:" or "likes:". However all rolls contain this (as shown in attachment) so the claimer ignores all roles. My solution was to simply remove this exception but one could also code it so that $im messages are detected in some other way.

This is the piece of code I removed from MudaeClaimer.cs :
` // ignore $im messages
            if (description.Any(l => l.StartsWith("claims:", StringComparison.OrdinalIgnoreCase) || l.StartsWith("likes:", StringComparison.OrdinalIgnoreCase)))
                return;`

Example of a normal roll containing "Claims:" and "Likes:" (in this server $ was replaced by " ) :
![image](https://user-images.githubusercontent.com/67010012/84785960-08026380-afe4-11ea-9f69-8d139cf0a34b.png)

**Comment by KingJackFrost** [comment]
> However all rolls contain this (as shown in attachment) so the claimer ignores all roles. 
I want to clarify this second point. I believe this setting depends on whether or not the server admin decides to show kakera values for characters. For my server, the "claims" and "likes" as well as kakera value do not show. Perhaps a small on/off enabler for this could help (such as a switch case).

**Comment by thebatsyjoke** [comment]
Yes, well noted. I did not remember there was such setting. Nonethelss, a fix like the one you suggested should be fairly simple to implement. Another possibility would be to check in the embed for a gender emoji, as these (I think) are only shown in $im messages.
I would also like to add that the bot checks for harem messages such as $mm. However, this does not affect its functionality and I don't believe there's any easy fix other than checking for the user's message content.

**Comment by WaterBottled** [comment]
How do i run mudaefarm from csharp after i remove the code? thanks for the help.

**Comment by thebatsyjoke** [comment]
You need to have Microsoft .NET installed. Make sure you download all the source code (I had to download Disqord folder separatedly). Open up a command prompt window and open the master folder with `cd` (where the MudaeClaimer.cs is located). Run the command `dotnet build` and wait for it to compile the application. You can now find the MudaeFarm.exe in "Master folder"\MudaeFarm\bin\Debug\netcoreapp3.1\win-x64\MudaeFarm.exe

**Comment by phosphene47** [comment]
I couldn't fix this, but I released [v4.0.4](https://github.com/chiyadev/MudaeFarm/releases/tag/v4.0.4) that contains a workaround.

**Comment by manuelsousa-21907715** [comment]
how to desactivate auto update? i actually want to go to 4.0.3 again

**Comment by manuelsousa-21907715** [comment]
nvm, i know already, u just need to close when is updating and put auto update false

**Comment by yud54** [comment]
> You need to have Microsoft .NET installed. Make sure you download all the source code (I had to download Disqord folder separatedly). Open up a command prompt window and open the master folder with `cd` (where the MudaeClaimer.cs is located). Run the command `dotnet build` and wait for it to compile the application. You can now find the MudaeFarm.exe in "Master folder"\MudaeFarm\bin\Debug\netcoreapp3.1\win-x64\MudaeFarm.exe

Where can I download the discord folder?

**Comment by thebatsyjoke** [comment]
The Disqord folder is located in the master branch of the project. However if I'm not mistaken, as of version 4.0.4 the bug as been patched.

**Comment by yud54** [comment]
one question again XD, how to logout discord account at mudaefarm ?

**Comment by thebatsyjoke** [comment]
I'm not sure I understood your question. If you mean to remove your current Discord account from MudaeFarm, you can delete the default profile within `%localappdata%\MudaeFarm\profiles.json`.

**Comment by yud54** [comment]
thank you, 
in 1 pc whether it can run 3 mudaefarm programs for 3 accounts in 1 pc?

**Comment by phosphene47** [comment]
> thank you,
> in 1 pc whether it can run 3 mudaefarm programs for 3 accounts in 1 pc?

Use profiles to run multiple instances of MudaeFarm with different accounts.

Also, please create a new issue if your question is irrelevant.

**Comment by yud54** [comment]
ahh sorry, I just created a github account, please understand, and sorry if my grammar is a mess I use google translate to translate it.
